rclcpp/include/rclcpp/parameter_descriptor_wrapper.hpp

```diff
@@ -0,0 +1,102 @@
+#ifndef RCLCPP__PARAMETER_DESCRIPTOR_WRAPPER_HPP_
+#define RCLCPP__PARAMETER_DESCRIPTOR_WRAPPER_HPP_
+
+// Standard library includes
+#include <functional>
+#include <utility>
+#include <memory>
+
+// Additional ROS libraries needed
+#include "node.hpp"
+#include "rcl_interfaces/msg/describe_parameters.hpp"
+#include "rclcpp/node.hpp"
+
+namespace rclcpp
+{
+
+// Implments ParameterDesription class with builder design pattern
+class ParameterDescription
+{
+public:
+    // List of classes the builder manages
+    class FloatingPointDescription;
+
+    // Two constructors in case certain methods don't need to declare_parameters with a Type T in the form of a template
+    ParameterDescription(std::string name, std::uint8_t type, std::string description, std::string additional_constraints, bool read_only. bool dynamic_typing) : m_name{name}, parameter_descriptor.type{type}, m_description{description}, m_additional_constraints{additional_constraints}, read_only{read_only}, dyanmic_typing{m_dyanmic_typing}{};
+
+private:
+    // The main descriptor object we're meant to adjust
+    rcl_interfaces::msg::ParameterDescriptor parameter_descriptor = {};
+
+    // Copies all the information in ParameterDescriptor.msg - https://github.com/ros2/rcl_interfaces/blob/rolling/rcl_interfaces/msg/ParameterDescriptor.msg
+    std::string m_name;
+    std::string m_description;
+
+    // Parameter Constraints
+    std::string m_additional_constraints;
+    bool m_read_only;
+    bool m_dynamic_typing;
+
+};
+
+class ParameterDescription::FloatingPointDescription
+{
+public:
+    // Our Main build methods which will construct the base class
+    ParameterDescription build() const;
+
+    //Builder Method - Mains
+    FloatingPointDescription& SetName(std::string name);
+    FloatingPointDescription& SetType(std::uint8_t type);
+    FloatingPointDescription& SetDescription(std::string description);
```

Comment:
This should be something like "set description text", because the whole object is a description